packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -372,6 +383,166 @@ abstract class RegularWindowController extends BaseWindowController {
   void setFullscreen(bool fullscreen, {Display? display});
 }
 
+/// Delegate class for dialog window controller.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+///  * [DialogWindowController], the controller that creates and manages dialog windows.
+///  * [DialogWindow], the widget for a dialog window.
+@internal
+mixin class DialogWindowControllerDelegate {
+  /// Invoked when the user attempts to close the window.
+  ///
+  /// The default implementation destroys the window. Subclasses
+  /// can override the behavior to delay or prevent the window from closing.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  /// * [onWindowDestroyed], which is invoked after the window is closed.
+  @internal
+  void onWindowCloseRequested(DialogWindowController controller) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+
+    controller.destroy();
+  }
+
+  /// Invoked after the window is closed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  ///
+  /// See also:
+  ///
+  /// * [onWindowCloseRequested], which is invoked when the user attempts to close the window.
+  @internal
+  void onWindowDestroyed() {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+}
+
+/// A controller for a dialog window.
+///
+/// Two types of dialogs are supported:
+///  * Modal dialogs: created with a non-null parent. These dialogs are modal
+///    to the parent, do not have a system menu, and are not selectable from the
+///    window switcher.
+///  * Modeless dialogs: created with a null parent. These dialogs can be
+///    minimized (but not maximized), and have a disabled close button.
+///
+/// This class does not interact with the widget tree. Instead, it is typically
+/// provided to the [DialogWindow] widget, which renders the content inside the
+/// dialog window.
+///
+/// The user of this class is responsible for managing the lifecycle of the window.
+/// When the window is no longer needed, the user should call [destroy] on this
+/// controller to release the resources associated with the window.
+///
+/// {@tool snippet}
+/// An example usage might look like:
+///
+/// ```dart
+/// // ignore_for_file: invalid_use_of_internal_member
```

Comment:
Don't forget to remove this when the feature flag is removed. Maybe put a TODO above this code block or something?